zerver/openapi/test_curl_examples.py

```diff
@@ -35,45 +37,67 @@ def test_generated_curl_examples_for_success(client: Client) -> None:
     # on "add" tests coming before "remove" tests in some cases.  We
     # should try to either avoid ordering dependencies or make them
     # very explicit.
-    for file_name in sorted(glob.glob("templates/zerver/api/*.md")):
-        with open(file_name) as f:
+    current_dir = os.path.dirname(os.path.abspath(__file__))
+    api_docs_file = os.path.join(
+        current_dir, "..", "..", "templates/zerver/help/include/rest-endpoints.md"
+    )
+    api_docs_list = open(api_docs_file, "r").read()
+    ENDPOINT_REGEXP = re.compile(r"api/\s*(.+?)\s*\)")
+    endpoint_list = sorted(re.findall(ENDPOINT_REGEXP, api_docs_list))
+    for endpoint in endpoint_list:
+        article_name = endpoint + ".md"
+        file_path = os.path.join(current_dir, "..", "..", "templates/zerver/api/", article_name)
```

Comment:
Why the change to use the `../../` stuff?  We previously were having paths starting from the CWD, which should be correct and save.

Reply:
@timabbott opened https://github.com/zulip/zulip/pull/18696 for this commit instead.